src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtposition/InstallmentProcessingService.java

```diff
@@ -0,0 +1,101 @@
+
+    public InstallmentProcessingService(DebtPositionService debtPositionService,
+                                        InstallmentSynchronizeMapper installmentSynchronizeMapper,
+                                        InstallmentErrorsArchiverService installmentErrorsArchiverService,
+                                        DPInstallmentsWorkflowCompletionService dpInstallmentsWorkflowCompletionService) {
+        this.debtPositionService = debtPositionService;
+        this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+        this.installmentErrorsArchiverService = installmentErrorsArchiverService;
+        this.dpInstallmentsWorkflowCompletionService = dpInstallmentsWorkflowCompletionService;
+    }
+
+    /**
+     * Processes a stream of InstallmentIngestionFlowFileDTO and synchronizes each installment.
+     *
+     * @param iterator          Stream of installment ingestion flow file DTOs to be processed.
+     * @param ingestionFlowFile Metadata of the ingestion file containing details about the ingestion process.
+     * @param workingDirectory  The directory where error files will be written if processing fails.
+     * @return An {@link InstallmentIngestionFlowFileResult} containing details about the processed rows, errors, and archived files.
+     */
+    public InstallmentIngestionFlowFileResult processInstallments(Iterator<InstallmentIngestionFlowFileDTO> iterator,
+                                                                  IngestionFlowFile ingestionFlowFile,
+                                                                  Path workingDirectory) {
+        List<InstallmentErrorDTO> errorList = new ArrayList<>();
+        long processedRows = 0;
+        long totalRows = 0;
+
+        while (iterator.hasNext()) {
+            totalRows++;
+
+            InstallmentIngestionFlowFileDTO installment = iterator.next();
+            InstallmentSynchronizeDTO installmentSynchronizeDTO = installmentSynchronizeMapper.map(
+                    installment,
+                    ingestionFlowFile.getIngestionFlowFileId(),
+                    ingestionFlowFile.getOrganizationId()
+            );
+            try {
+                // For the moment we have decided to call the GPD api for a single DP because their development for the massive v2.0 is not yet ready
+                String workflowId = debtPositionService.installmentSynchronize("ORDINARY_SIL", installmentSynchronizeDTO, false);
+                if (workflowId != null) {
+                    boolean workflowCompleted = dpInstallmentsWorkflowCompletionService.waitForWorkflowCompletion(
+                            workflowId, installment, ingestionFlowFile.getFileName(), errorList
+                    );
+                    log.info("Workflow with id {} completed", workflowId);
+                    if (workflowCompleted) {
+                        processedRows++;
+                    }
+                } else {
```

Comment:
move these rows into a separate `waitWorkflowCompletion` method, which will return true if the row has been processed (if worflowId was null or the workflow completed sucessfully).
The you will update processedRows once if the method returned true